cli/azd/cmd/testdata/TestUsage-azd-pipeline-config.snap

```diff
@@ -9,15 +9,16 @@ Usage
   azd pipeline config [flags]
 
 Flags
-        --auth-type string           	: The authentication type used between the pipeline provider and Azure for deployment (Only valid for GitHub provider). Valid values: federated, client-credentials.
-        --docs                       	: Opens the documentation for azd pipeline config in your web browser.
-    -e, --environment string         	: The name of the environment to use.
-    -h, --help                       	: Gets help for config.
-        --principal-id string        	: The client id of the service principal to use to grant access to Azure resources as part of the pipeline.
-        --principal-name string      	: The name of the service principal to use to grant access to Azure resources as part of the pipeline.
-        --principal-role stringArray 	: The roles to assign to the service principal. By default the service principal will be granted the Contributor and User Access Administrator roles.
-        --provider string            	: The pipeline provider to use (github for Github Actions and azdo for Azure Pipelines).
-        --remote-name string         	: The name of the git remote to configure the pipeline to run on.
+        --auth-type string                	: The authentication type used between the pipeline provider and Azure for deployment (Only valid for GitHub provider). Valid values: federated, client-credentials.
+        --client-secret-duration duration 	: The duration for which the client secret should be valid. The default is 180 days.
```

Comment:
would really like not to have this and think about deprecating client-secret instead.

Adding this flag brings some ambiguity when combining `auth-type=federated` with it, or provider github with it.  There would be more invalid cases to catch.  Like, what if I set a million days?

Reply:
Did not see an ask on deprecating the usage of SP's using client-secrets/passwords. Still think we should be supporting it.

Reply:
@rajeshkamal5050 , why should we support it?  We already have federated as default for github. 
I would understand if `federated credential` was something which might be blocked for some users or companies, so they need some alternative; but that's not the case...  
So, I wonder what would it be the point of offering a less-secure alternative here...

Reply:
While the push right now is about improving our usages, we also need to make sure our customers are being more secure by default as well. If there are better alternatives, we should default to those and only keep support for SP's if that is the only option.

Reply:
> If there are better alternatives, we should default to those and only keep support for SP's if that is the only option.

We do this today - on GitHub, which supports federated credentials, we default to using that, which means there is no shared secrets.  We want to move to that for AzDo pipelines as well.